### PR TITLE
Convert IronMeta utility class into an ES6 class

### PR DESCRIPTION
**Pull request by aomarks**
This allows Analyzer to understand the shape of this class.

### CODE REVIEW

**Comment by e111077** [comment]
I'm not comfortable reviewing iron-meta. I think we'll have to wait for Russell to get back tomorrow

**Comment by aomarks** [comment]
> I'm not comfortable reviewing iron-meta. I think we'll have to wait for Russell to get back tomorrow

Ok. Note this is inert code -- it just adds type annotations.

**Comment by aomarks** [comment]
> Ok. Note this is inert code -- it just adds type annotations.

Actually ended up needing a more substantial change. I converted the `IronMeta` class (not the element, the utility class) into an ES6 class.

File: iron-meta.js [hunk 1]
```diff
@@ -74,14 +72,25 @@ IronMeta.prototype = {
         return metaDatas[this.type][key];
       }, this);
     }
-  },
+  }
 
-  byKey: function(key) {
+  /**
+   * @param {string} key
```
Comment (bicknellr) [inline iron-meta.js]: Should this match the annotations above with `{string|null|undefined}`?

Comment (aomarks) [reply iron-meta.js]: if key is not truthy you always get undefined, so it seems to make more sense for this method to require you provide a key

**Pull request merged**